src/providers/EtherspotContextProvider.tsx

```diff
@@ -540,6 +547,51 @@ const EtherspotContextProvider = ({
     [sdk, accountAddress]
   );
 
+  const getTransactionsFromChain = useCallback(
+    async (chainId: number, address: string | null): Promise<Transactions[]> => {
+      const sdk = getSdkForChainId(chainId);
+      const account = address;
+      if (!!sdk && !account) return [];
+
+      await sdk.computeContractAccount();
+      try {
+        let transactions: any = [];
+        const txs = await sdk.getTransactions({ account });
+        if (txs?.items) transactions = txs.items?.map((item) => ({ ...item, chainId }));
+        return { transactions };
+      } catch (e) {
+        //
+      }
+
+      return [];
+    },
+    [sdk, accountAddress]
+  );
+
+  const getAllTransactions = useCallback(
+    async (address: string | null): Promise<IAllChainTransactions[]> => {
+      let tempTransactions: IAllChainTransactions = {};
+      let tempTransactionsChanges = 0;
+
+      await Promise.all(
+        supportedChains.map(async (chain) => {
+          try {
+            let transactionsResult = await getTransactionsFromChain(chain.chainId, address);
+            let chain_id = chain.chainId;
+            if (transactionsResult?.transactions) {
+              tempTransactions[chain_id] = transactionsResult.transactions;
+              tempTransactionsChanges++;
+            }
+          } catch (e) {
+            //
```

Comment:
Another empty catch block, please always handle the negative scenario

Reply:
same comment as above.
I am calling  let transactionsResult = await getTransactionsFromChain(chain.chainId, address); inside this function. and handled blank array if no TX and any error, so I return tmp array blank here. after catch.

Reply:
This is less of a problem as the issue above, but you need to at least handle the possibilty of the `.catch` block firing, otherwise there's no point using a try / catch block in the first place